src/hooks/async/contracts.ts

```diff
@@ -349,3 +353,47 @@ export function useContractFunctions(
     },
   );
 }
+
+export function useContractData(
+  contract: RequiredParam<ReturnType<typeof useContract>["contract"]>,
+  functionName: RequiredParam<string>,
+  ...args: unknown[]
```

Comment:
args have to be `unknown[]` and also optionally allow  the call overrides as a last param (see how typescript sdk does this here: https://github.com/thirdweb-dev/typescript-sdk/blob/7ae776e7b94dc75ea7032e79a1a0c94d4414cf1f/src/contracts/smart-contract.ts#L182